### PR TITLE
Add an example app for the EFR32

### PR DESCRIPTION
**Pull request by sagar-apple**
#### Problem
No way to exercise the EFR32 platform specific code. 

 #### Summary of Changes
Add an example app for the EFR32 with instructions on how to build it. 
Currently the CHIP stack is very limited but this app is able to launch with CHIP linked and use some of the features that are implemented for now. 

We are working with the SDK authors on making this app easier to build.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]
Approving pending comments, but LGTM

**Review by hawk248 — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Review by suryanshup — APPROVED** [review]

**Review by gerickson — APPROVED** [review]

**Pull request merged**
